app/frontend/src/features/donations/DonationsBoxMixed.tsx

```diff
@@ -0,0 +1,458 @@
+    handleSubmit,
+    reset: resetForm,
+    errors,
+  } = useForm<DonationFormData>({ defaultValues: { recurring: true } });
+
+  const {
+    error,
+    isLoading,
+    mutate: initiateDonation,
+  } = useMutation<void, GrpcError, DonationFormData>(
+    async (formData) => {
+      const stripe = (await stripePromise)!;
+      const session_id = await service.donations.initiateDonation(formData);
+      // When the customer clicks on the button, redirect them to Checkout.
+      const result = await stripe.redirectToCheckout({
+        sessionId: session_id,
+      });
+      if (result.error) {
+        throw Error(result.error.message);
+      }
+    },
+    {
+      onSuccess: () => {
+        resetForm();
+      },
+    }
+  );
+
+  const onSubmit = handleSubmit((data) => {
+    initiateDonation(data);
+  });
+
+  return (
+    <>
+      <form onSubmit={onSubmit} className={classes.donationsBox}>
+        {error && <Alert severity="error">{error.message}</Alert>}
+        {success && (
+          <Alert severity="success">
+            Thank you for donating to Couchers.org! We appreciate your support!
+          </Alert>
+        )}
+        {cancelled && (
+          <Alert severity="warning">The payment was cancelled.</Alert>
+        )}
+        <Typography className={classes.marginBottom2} variant="h3">
+          {DONATIONSBOX_TITLE}
+        </Typography>
+        <Controller
+          id="recurring"
+          control={control}
+          name="recurring"
+          rules={{ required: "This field is required." }}
+          defaultValue="monthly"
+          render={({ onChange, value }) => (
+            <FormControl className={classes.formGroup}>
+              <RadioGroup
+                className={classes.donationsBoxRow}
+                aria-label="recurring"
+                name="recurring-radio"
+                onChange={(e) => onChange(e.target.value === "monthly")}
+                value={value ? "monthly" : "one-time"}
+              >
+                <FormControlLabel
+                  className={classes.buttonSecondaryRadio}
+                  value="monthly"
+                  control={<Radio />}
+                  label={DONATIONSBOX_MONTHLY}
+                />
+                <FormControlLabel
+                  className={classes.buttonSecondaryRadio}
+                  value="one-time"
+                  control={<Radio />}
+                  label={DONATIONSBOX_ONETIME}
+                />
+              </RadioGroup>
+              <FormHelperText error={!!errors?.recurring?.message}>
+                {errors?.recurring?.message ??
+                  "A monthly gift helps us plan ahead and build our operations on a stable footing."}
+              </FormHelperText>
+            </FormControl>
+          )}
+        />
+
+        <Divider className={classes.marginY2} />
+
+        <div className={classes.donationsBoxRow}>
+          <div className={classes.donationsBoxSubRow}>
+            <button
```

Comment:
all these buttons need `type="button"`, they currently submit the form

If you feel really excited, I think you could reduce some boilerplate but having all these buttons as `<radio>`s, but that's totally optional :P

Reply:
I also thought about making them radio, but then I realized that we have the "extra" text input. As I understand it, when we set a custom value, the rest of the values should be unselected (if not it would be very confusing), so that is kind of not consistent with them being radio (as radio buttons should always have one of them checked).